src/protocol/wrapper_generator.cpp

```diff
@@ -385,6 +385,20 @@ class Interface
             emit_indented_lines(out, indent + "    ",
                 {{"wl_resource_set_implementation(resource, get_vtable(), me, nullptr);"}});
         }
+        emit_indented_lines(out, indent + "    ", {
+            {"try"},
+            {"{"},
+            {"  me->bind(client, resource);"},
+            {"}"},
+            {"catch(...)"},
+            {"{"},
+            {"    ::mir::log("},
+            {"        ::mir::logging::Severity::critical,"},
```

Comment:
Hmm, I see this isn't the only instance around here, but I don't think an exception is "Severity::critical" if we swallow it and continue running.

Anyway. I think this is at most "Severity::warning". (If it is really "Severity::critical" we should be closing the process asap.)